### PR TITLE
[test] packaging: gradle tasks for groovy tests

### PR DESCRIPTION
**Pull request by andyb-elastic**
The vagrant test plugin adds tasks for the groovy packaging tests,
which run after the bats packaging test tasks.Rename the 'bats'
configuration to 'packaging' and remove the option to inherit
archives from this configuration.

For #26741

This originally appeared in #27623

### CODE REVIEW

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy [hunk 1]
```diff
@@ -154,20 +156,39 @@ class VagrantTestPlugin implements Plugin<Project> {
     }
 
     private static void createPrepareVagrantTestEnvTask(Project project) {
-        File batsDir = new File("${project.buildDir}/${BATS}")
+        File packagingDir = new File(project.buildDir, PACKAGING_CONFIGURATION)
+        Task createPackagingDirTask = project.tasks.create('creatingPackagingDir')
```
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy]: Why do we need this task? Copy tasks will create the destination dir if it doesn't exist. Worst case, this could be a doFirst on copyPackagingArchives.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy [hunk 2]
```diff
@@ -359,32 +368,41 @@ class VagrantTestPlugin implements Plugin<Project> {
...
-                project.gradle.removeListener(packagingReproListener)
+            if (project.extensions.esvagrant.boxes.contains(box)) {
+                packagingTest.dependsOn(batsPackagingTest)
+            }
+
+            Task groovyPackagingTest = project.tasks.create("vagrant${boxTask}#groovyPackagingTest")
+            groovyPackagingTest.dependsOn(up)
+            groovyPackagingTest.finalizedBy(halt)
+            groovyPackagingTest.doLast {
+                // no op
```
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy]: Why is this needed?

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy [hunk 3]
```diff
@@ -415,4 +425,16 @@ class VagrantTestPlugin implements Plugin<Project> {
...
         }
     }
+
+    private static TaskExecutionAdapter createReproListener(Project project) {
+        return new TaskExecutionAdapter() {
+            @Override
+            void afterExecute(Task task, TaskState state) {
+                final String gradlew = Os.isFamily(Os.FAMILY_WINDOWS) ? "gradlew" : "./gradlew"
+                if (state.failure != null) {
+                    println "REPRODUCE WITH: ${gradlew} ${task.path} -Dtests.seed=${project.testSeed} "
```
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy]: Using `task.path` only works because this listener is always be added/removed in doFirst/doLast on a task. Rather that depend on that behavior, why not pass in the task name to `createReproListener`?

**Review by rjernst — COMMENTED** [review]
I left a few comments.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy (hunk 3 above)
Comment (andyb-elastic) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy]: Gotcha - my thinking was to prevent the repro message from mistakenly having a different task than the one that triggered the listener. But since the listener is added to the project, not the task, that makes it less clear what's happening. I'll have it pass the task as a parameter

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy (hunk 2 above)
Comment (andyb-elastic) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy]: I added the empty groovyPackagingTest task to introduce changes incrementally (the next PR will make this into a `VagrantCommandTask` that executes the test jar)

The empty `doLast` with the comment is just to make it explicit that the task doesn't currently do anything

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy (hunk 1 above)
Comment (andyb-elastic) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy]: You're right, we don't. I inferred from the `createBatsDirs` task that it had to already exist - I'll remove both

**Comment by andyb-elastic** [comment]
@rjernst I pushed https://github.com/andyb-elastic/elasticsearch/commit/b79c10409b1622c2931f8cda5d4f648c59c8f3a3 to address your comments

**Comment by andyb-elastic** [comment]
Jenkins run packaging tests please

**Review by rjernst — APPROVED** [review]
LGTM
